src/librustc_borrowck/diagnostics.rs

```diff
@@ -314,6 +314,76 @@ let c = &i; // still ok!
 ```
 "##,
 
+E0501: r##"
+This error indicates that a mutable variable which has been captured in a
+closure is being used while it is still captured by the closure. Because the
+closure has borrowed the variable, the variable is not available for use until
+the closure goes out of scope.
+
+Example of erroneous code:
+```compile_fail
+struct Sample;
+fn inside_closure(x: &mut Sample) {
+    // Actions which require unique access
+}
+fn outside_closure(x: &mut Sample) {
+    // Actions which require unique access
+}
+fn foo(a: &mut Sample){
+    let mut bar = || {
+        inside_closure(a)
+    };
+    outside_closure(a); // error: cannot borrow `*a` as mutable because previous
+                        //        closure requires unique access.
+}
+```
+
+To fix this error, you can place the closure in it's own environment:
```

Comment:
Yep. I realized the exception immediately after I posted the second comment.  Thanks for the review.